analytics/http/http_module.go

```diff
@@ -0,0 +1,274 @@
+	}
+
+	shouldTrackCookieSync, err := createCookieSyncFilter(cfg.CookieSync)
+	if err != nil {
+		return nil, err
+	}
+
+	shouldTrackNotification, err := createNotificationFilter(cfg.Notification)
+	if err != nil {
+		return nil, err
+	}
+
+	shouldTrackSetUID, err := createSetUIDFilter(cfg.SetUID)
+	if err != nil {
+		return nil, err
+	}
+
+	shouldTrackVideo, err := createVideoFilter(cfg.Video)
+	if err != nil {
+		return nil, err
+	}
+
+	// init buffer
+	buffer := bytes.Buffer{}
+	buffer.Write([]byte("["))
+
+	// Check for filters
+	return &HttpLogger{
+		sender: sender,
+		clock:  clock,
+
+		maxBufferByteSize: pSize,
+		eventCount:        0,
+		maxEventCount:     int64(cfg.Buffers.EventCount),
+		maxDuration:       pDuration,
+
+		// Filters
+		shouldTrackAuction:      shouldTrackAuction,
+		shouldTrackAmp:          shouldTrackAmp,
+		shouldTrackCookieSync:   shouldTrackCookieSync,
+		shouldTrackNotification: shouldTrackNotification,
+		shouldTrackSetUID:       shouldTrackSetUID,
+		shouldTrackVideo:        shouldTrackVideo,
+
+		buffer:    buffer,
+		bufferCh:  make(chan []byte),
+		sigTermCh: make(chan os.Signal),
+	}, nil
+}
+
+func NewModule(httpClient *http.Client, cfg config.AnalyticsHttp, clock clock.Clock) (analytics.Module, error) {
+	sender, err := createHttpSender(httpClient, cfg.Endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	m, err := newHttpLogger(cfg, sender, clock)
+	if err != nil {
+		return nil, err
+	}
+
+	signal.Notify(m.sigTermCh, os.Interrupt, syscall.SIGTERM)
+
+	go m.start()
+
+	return m, nil
+}
+
+func (l *HttpLogger) start() {
+	ticker := l.clock.Ticker(l.maxDuration)
+	for {
+		select {
+		case <-l.sigTermCh:
+			glog.Infof("[HttpAnalytics] Received Close, trying to flush buffer")
+			l.flush()
+			return
+		case event := <-l.bufferCh:
+			l.bufferEvent(event)
+			if l.isFull() {
+				l.flush()
+			}
+		case <-ticker.C:
+			l.flush()
+		}
+	}
+}
+
+func (l *HttpLogger) bufferEvent(data []byte) {
+	l.mux.Lock()
+	defer l.mux.Unlock()
+
+	l.buffer.Write(data)
+	l.buffer.WriteByte(',')
+	l.eventCount++
+}
+
+func (l *HttpLogger) isFull() bool {
+	l.mux.RLock()
+	defer l.mux.RUnlock()
+	return l.eventCount >= l.maxEventCount || int64(l.buffer.Len()) >= l.maxBufferByteSize
+}
+
+func (l *HttpLogger) flush() {
+	if l.eventCount == 0 || l.buffer.Len() == 0 {
+		return
+	}
+	l.mux.Lock()
+	defer l.reset()
+	defer l.mux.Unlock()
+
+	// Close the json array, remove last ,
+	l.buffer.Truncate(l.buffer.Len() - 1)
+	_, err := l.buffer.Write([]byte("]"))
+	if err != nil {
+		glog.Warning("[HttpAnalytics] fail to close the json array")
+		return
+	}
+	payload := make([]byte, l.buffer.Len())
+	_, err = l.buffer.Read(payload)
+	if err != nil {
+		glog.Warning("[HttpAnalytics] fail to copy the buffer")
+		return
+	}
+	go l.sender(payload)
+}
+
+func (l *HttpLogger) reset() {
+	l.buffer.Reset()
+	l.buffer.Write([]byte("["))
+	l.eventCount = 0
+}
+
+func (m *HttpLogger) LogAuctionObject(event *analytics.AuctionObject) {
```

Comment:
Im interested in why this variable name has changed from `l` to `m` in relation to the httpLogger?